diff --git a/cmake/opencv.cmake b/cmake/opencv.cmake
index 971854b..794f818 100644
--- a/cmake/opencv.cmake
+++ b/cmake/opencv.cmake
@@ -34,7 +34,7 @@ if(WIN32)
                   "opencv_videoio481$<$<CONFIG:Debug>:d>")
   if("${CMAKE_CXX_COMPILER_ID}" MATCHES "MSVC")
     if("${MSVC_VERSION}" LESS "1940")
-      set(OPENCV_STATICLIB_PREFIX "x64/vc17/")
+      set(OPENCV_STATICLIB_PREFIX "x64/vc16/")
     else()
       set(OPENCV_STATICLIB_PREFIX "/")
     endif()
diff --git a/src/alpha_omega_node.cpp b/src/alpha_omega_node.cpp
index e2d1ba8..69cc792 100644
--- a/src/alpha_omega_node.cpp
+++ b/src/alpha_omega_node.cpp
@@ -818,7 +818,7 @@ namespace thalamus {
           impl->GetChannelsCount([this] {
             impl->GetAllChannels([this] {
               impl->AddBufferChannels(impl->channel_ids, [this] {
-                size_t polling_interval = impl->state->at("Polling Interval");
+                long long polling_interval = impl->state->at("Polling Interval");
                 impl->current_band = 0;
                 impl->last_frequency_update = std::chrono::steady_clock::now();
                 impl->on_timer(boost::system::error_code(), std::chrono::milliseconds(polling_interval));
diff --git a/src/base_node.cpp b/src/base_node.cpp
index 17409be..f58bc90 100644
--- a/src/base_node.cpp
+++ b/src/base_node.cpp
@@ -24,7 +24,7 @@ namespace thalamus {
     double threshold;
     bool _on;
     double last_sample;
-    size_t channel = 0;
+    long long channel = 0;
 
     Impl(ObservableDictPtr state, NodeGraph* graph)
       : state(state)
@@ -326,7 +326,7 @@ namespace thalamus {
     bool high = false;
     double last_sample = 0;
     std::chrono::nanoseconds _time;
-    size_t channel = 0;
+    long long channel = 0;
     ToggleNode* outer;
     std::chrono::nanoseconds last_toggle;
     std::chrono::nanoseconds current_time;
diff --git a/src/channel_picker_node.cpp b/src/channel_picker_node.cpp
index c74802e..cac57a3 100644
--- a/src/channel_picker_node.cpp
+++ b/src/channel_picker_node.cpp
@@ -64,7 +64,7 @@ namespace thalamus {
         if (!mapping_dict->contains("Out Channel")) {
           return;
         }
-        size_t out_channel = mapping_dict->at("Out Channel");
+        long long out_channel = mapping_dict->at("Out Channel");
         if (mappings.size() < out_channel + 1) {
           mappings.resize(out_channel + 1);
         }
diff --git a/src/nidaq_node_windows.cpp b/src/nidaq_node_windows.cpp
index 0e73951..637e012 100644
--- a/src/nidaq_node_windows.cpp
+++ b/src/nidaq_node_windows.cpp
@@ -299,7 +299,7 @@ namespace thalamus {
 
           _sample_interval = std::chrono::nanoseconds(size_t(1e9 / sample_rate));
 
-          size_t polling_interval_raw = state->at("Poll Interval");
+          long long polling_interval_raw = state->at("Poll Interval");
           std::chrono::milliseconds polling_interval(polling_interval_raw);
 
           _every_n_samples = polling_interval / _sample_interval;
diff --git a/src/xsens_node.cpp b/src/xsens_node.cpp
index b717f6b..dea5da0 100644
--- a/src/xsens_node.cpp
+++ b/src/xsens_node.cpp
@@ -96,7 +96,7 @@ namespace thalamus {
     std::span<Segment const> _segment_span;
     bool is_running = false;
     unsigned char buffer[4096];
-    size_t port;
+    long long port;
     XsensNode* outer;
     std::chrono::nanoseconds time;
     std::string pose_name;
